### ISSUE TITLE
User Daemon hangs on starts

### ISSUE THREAD

**Issue by Gikkman** [telepresenceio/telepresence#2377]
**Describe the bug**
When I run `telepresence connect`, the root daemon starts, but then the program hangs on creating the user daemon:
```
% telepresence loglevel debug         

Launching Telepresence Root Daemon
Need root privileges to run: /opt/homebrew/bin/telepresence daemon-foreground /Users/simon.jonasson/Library/Logs/telepresence '/Users/simon.jonasson/Library/Application Support/telepresence' ''
Password:
Launching Telepresence User Daemon
[[Here I have to CTRL+C]]

% telepresence status                                                                                                                                                                                                     
Root Daemon: Running
  Version   : v2.4.11 (api 3)
  DNS       :
    Remote IP       : 172.30.0.10
    Exclude suffixes: [.arpa .com .io .net .org .ru]
    Include suffixes: []
    Timeout         : 4s
  Also Proxy : (0 subnets)
  Never Proxy: (1 subnets)
User Daemon: Running
[[Here I have to CTRL+C again]]

% telepresence quit                                                                                                                                                                                                       
Telepresence Root Daemon quitting... done
Telepresence User Daemon quitting... done
```

**Versions**
```
% telepresence version                                        
Client: v2.4.11 (api v3)
Root Daemon: v2.4.11 (api v3)
User Daemon: v2.4.11 (api v3)
```
Laptop: MacOS 12.1 - Apple M1 Pro
Cluster: Openshift (Kubernetes 1.21.1, Openshift 4.8)

**Additional context**
The interesting thing is that it seems that everything works as it should. I can still run `curl` against services using the format `service.namespace`. So I guess stuff is working, it's just the command that hangs.

Attaching logs:
[logs.zip](https://github.com/telepresenceio/telepresence/files/8048631/logs.zip)

**Comment by thallgren** [comment]
One thing that I notice by looking at the lots is that your traffic manager is of version 2.4.6. In general, this _should_ work, but it might be a good idea to upgrade.

**Comment by thallgren** [comment]
Also, you might get some more info if you start with `telepresence loglevel debug`, before doing connect.

**Comment by Gikkman** [comment]
> Also, you might get some more info if you start with `telepresence loglevel debug`, before doing connect.

That's what I did when I collected these logs. It starts both daemons when I execute that command, and the user daemon still hangs

> One thing that I notice by looking at the lots is that your traffic manager is of version 2.4.6. In general, this should work, but it might be a good idea to upgrade.

Can I really do that? Our entire company uses telepresence, and different devs might have different versions installed. If I upgrade the traffic manager, will it work with older version of telepresence? Or will they get a "please update" message when they try to connect?

**Comment by Gikkman** [comment]
Some updates:

We updated out Traffic Manager to `2.4.10`. I did notice some messages in the logs of the Traffic Manager, might be of importance?

```
2022-02-16 15:01:58.9543 info    Logging at this level "info"
2022-02-16 15:01:58.9544 info    Traffic Manager v2.4.10 [pid:1]
2022-02-16 15:01:58.9781 info    Using DNS IP from dns-default.openshift-dns
2022-02-16 15:01:58.9841 info    Using cluster domain "cluster.local."
2022-02-16 15:01:58.9859 info    Extracting service subnet 172.30.0.0/16 from create service error message
2022-02-16 15:01:58.9860 info    Using podCIDRStrategy: auto
2022-02-16 15:01:58.9864 info    agent-injector : Mutating webhook service is listening on :8443
2022-02-16 15:01:59.0865 info    Scanning 44 nodes
2022-02-16 15:01:59.0866 info    No subnets found
2022-02-16 15:01:59.0866 error   Unable to derive subnets from nodes
2022-02-16 15:02:03.6010 info    Deriving subnets from IPs of pods
2022-02-16 15:03:50.6554 error   httpd/conn=[::1]:8081 :    CONN udp 10.128.0.0:57621 -> 10.128.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:03:50.7328 error   httpd/conn=[::1]:8081 :    CONN udp 10.129.0.0:57621 -> 10.129.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:03:50.8710 error   httpd/conn=[::1]:8081 :    CONN udp 10.131.0.0:57621 -> 10.131.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:04:28.1706 error   httpd/conn=[::1]:8081 :    CONN udp 10.128.0.0:57621 -> 10.128.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:04:28.2467 error   httpd/conn=[::1]:8081 :    CONN udp 10.129.0.0:57621 -> 10.129.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:04:28.3109 error   httpd/conn=[::1]:8081 :    CONN udp 10.130.0.0:57621 -> 10.130.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:04:28.3867 error   httpd/conn=[::1]:8081 :    CONN udp 10.131.0.0:57621 -> 10.131.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:04:28.4623 error   httpd/conn=[::1]:8081 :    CONN udp 172.30.0.0:57621 -> 172.30.255.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:05:14.5157 error   httpd/conn=[::1]:8081 :    CONN udp 10.128.0.0:57621 -> 10.128.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:05:14.5880 error   httpd/conn=[::1]:8081 :    CONN udp 10.129.0.0:57621 -> 10.129.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:05:14.7268 error   httpd/conn=[::1]:8081 :    CONN udp 10.131.0.0:57621 -> 10.131.31.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:05:14.8066 error   httpd/conn=[::1]:8081 :    CONN udp 172.30.0.0:57621 -> 172.30.255.255:57621 stream-to-conn loop ended because context canceled
2022-02-16 15:06:00.1711 error   agent-injector : http: TLS handshake error from 10.130.0.1:43666: EOF
2022-02-16 15:06:15.3784 error   httpd/conn=[::1]:8081 :    CONN tcp 172.30.0.0:51995 -> 172.30.176.39:8080 stream-to-conn loop ended because context canceled
2022-02-16 15:07:22.4174 error   httpd/conn=[::1]:8081 :    CONN tcp 172.30.0.0:52012 -> 172.30.255.83:8080 stream-to-conn loop ended because context canceled
2022-02-16 15:07:24.4238 error   httpd/conn=[::1]:8081 :    CONN tcp 172.30.0.0:52014 -> 172.30.176.39:8080 stream-to-conn loop ended because context canceled
2022-02-16 15:22:00.1620 error   agent-injector : http: TLS handshake error from 10.130.0.1:47462: EOF
```

Attaching logs again. I tried starting the connection with `telepresence loglevel debug` but it tries to start connecting when I use that command. Same with `telepresence gather-logs`, it also tries to start connecting. So I zipped the logs manually:
[logs.zip](https://github.com/telepresenceio/telepresence/files/8081338/logs.zip)

**Comment by thallgren** [comment]
You seem to have many namespaces. Please try to limit the namespaces that Telepresence must create DNS entries for by using `telepresence connect --mapped-namespaces <comma separated list of namespaces>`.

**Comment by Gikkman** [comment]
Using `--mapped-namespaces` doesn't seem to have any effect on the User Daemon for me. It still hangs the command on startup.

**Comment by Gikkman** [comment]
This seems to have been resolved with the latest telepresence release (2.5.2). After an upgrade, my user daemon starts as intended. Whatever you did, seems to have done the trick. Closing this ticket.
